plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.php

```diff
@@ -148,48 +146,63 @@ public function createSubmissionFileNode($doc, $submissionFile)
             $revisionNode->setAttribute('extension', pathinfo($revision->path, PATHINFO_EXTENSION));
+            $submissionFileNode->appendChild($revisionNode);
 
-            if (array_key_exists('no-embed', $this->opts)) {
-                $hrefNode = $doc->createElementNS($deployment->getNamespace(), 'href');
-                if (array_key_exists('use-file-urls', $this->opts)) {
-                    $stageId = Repo::submissionFile()->getWorkflowStageId($submissionFile);
-                    $dispatcher = Application::get()->getDispatcher();
-                    $request = Application::get()->getRequest();
-                    $params = [
-                        'submissionFileId' => $submissionFile->getId(),
-                        'submissionId' => $submissionFile->getData('submissionId'),
-                        'stageId' => $stageId,
-                    ];
-                    $url = $dispatcher->url($request, PKPApplication::ROUTE_COMPONENT, $context->getPath(), 'api.file.FileApiHandler', 'downloadFile', null, $params);
-                    $hrefNode->setAttribute('src', $url);
-                } else {
-                    $hrefNode->setAttribute('src', $revision->path);
-                }
-                $hrefNode->setAttribute('mime_type', $revision->mimetype);
-                $revisionNode->appendChild($hrefNode);
-            } else {
+            if (!($this->opts['no-embed'] ?? false)) {
                 $embedNode = $doc->createElementNS($deployment->getNamespace(), 'embed', base64_encode(file_get_contents($localPath)));
                 $embedNode->setAttribute('encoding', 'base64');
                 $revisionNode->appendChild($embedNode);
+                continue;
             }
 
-            $submissionFileNode->appendChild($revisionNode);
+            $hrefNode = $doc->createElementNS($deployment->getNamespace(), 'href');
+            $revisionNode->appendChild($hrefNode);
+            $hrefNode->setAttribute('mime_type', $revision->mimetype);
+
+            if (!($this->opts['use-file-urls'] ?? false)) {
+                $hrefNode->setAttribute('src', $revision->path);
+                continue;
+            }
+
+            $baseParams ??= [
+                'submissionFileId' => $submissionFile->getId(),
+                'submissionId' => $submissionFile->getData('submissionId'),
+                'stageId' => Repo::submissionFile()->getWorkflowStageId($submissionFile),
+            ];
+            $params = $baseParams + ['fileId' => $revision->fileId];
+            $dispatcher ??= Application::get()->getDispatcher();
+            $request ??= Application::get()->getRequest();
+            $url = $dispatcher->url($request, PKPApplication::ROUTE_COMPONENT, $context->getPath(), 'api.file.FileApiHandler', 'downloadFile', null, $params);
+            $hrefNode->setAttribute('src', $url);
         }
 
+        // Report if no revision has been added
+        if (!$hasRevision) {
```

Comment:
I would change this parameter's name to indicate that the `hasRevision` does not refer to whether the SubmissionFile has any revisions, but whether we are adding any revisions to the export (this is the case, right?)

Reply:
Exactly, I'll think about a better name 😅